packages/@ourworldindata/grapher/src/controls/entityPicker/EntityPicker.tsx

```diff
@@ -683,8 +687,32 @@ class PickerOption extends React.Component<PickerOptionProps> {
         const { entityName, plotValue, formattedValue } = optionWithMetricValue
         const metricValue = formattedValue === entityName ? "" : formattedValue // If the user has this entity selected, don't show the name twice.
 
+        const isFlipping = (selectedEntityNames?: {
+            previous?: string[]
+            current?: string[]
+        }): boolean => {
+            const { previous = [], current = [] } = selectedEntityNames ?? {}
+            const animatedEntityName = difference(
+                union(previous, current),
+                intersection(previous, current)
+            )[0]
+            return animatedEntityName === entityName
+        }
+
         return (
-            <Flipped flipId={entityName} translate opacity>
+            <Flipped
+                flipId={entityName}
+                translate
+                opacity
+                onStart={(element, decisionData) => {
+                    if (isFlipping(decisionData)) {
+                        element.classList.add("animating")
```

Comment:
Setting the class this way is, sadly, a bit fickle:
The classname will be reset the next time the React tree renders - which is, for example, the case as soon as you hover over _any_ other PickerEntry.

An alternative could be to set `isAnimating` as a React state, or alternatively to pass it in as another prop from the parent somehow.

Reply:
Good catch!